components/centraldashboard/app/api_contributors.ts

```diff
@@ -0,0 +1,184 @@
+const EMAIL_RGX = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
+
+/**
+ * Handles an exception in an async block and converts it to a JSON
+ * response sent back to client
+ */
+// tslint:disable-next-line: no-any
+const surfaceProfileControllerErrors = (info: {res: Response, msg: string, err: any}) => {
+    const {res, msg, err} = info;
+    const code = (err.response && err.response.statusCode) || 400;
+    const error = err.body || msg;
+    console.log(`${msg} ${error}`);
+    apiError({res, code, error});
+};
+
+/**
+ * Given an owned namespace, list all contributors under it
+ * @param namespace Namespace to find contributors for
+ * @param profilesService The instance of DefaultApi to use
+ */
+const getContributors = async (namespace: string, profilesService: DefaultApi) => {
+    const {body} = await profilesService
+        .readBindings(undefined, namespace);
+    const users = mapWorkgroupBindingToSimpleBinding(body.bindings)
+        .filter((b) => b.role === 'contributor')
+        .map((b) => b.user);
+    return users;
+};
+
+const handleContributor = async (action: CONTRIBUTOR_ACTIONS, profilesService: DefaultApi, req: Request, res: Response) => {
+    const {namespace} = req.params;
+    const {contributor} = req.body as AddOrRemoveContributorRequest;
+    if (!contributor || !namespace) {
+        return apiError({
+            res,
+            error: `Missing contributor / namespace fields.`,
+        });
+    }
+    if (!EMAIL_RGX.test(contributor)) {
+        return apiError({
+            res,
+            error: `Contributor doesn't look like a valid email address`,
+        });
+    }
+    let errIndex = 0;
+    try {
+        const binding = mapSimpleBindingToWorkgroupBinding({
+            user: contributor,
+            namespace,
+            role: 'contributor',
+        });
+        const {headers} = req;
+        delete headers['content-length'];
+        const actionAPI = action === 'create' ? 'createBinding' : 'deleteBinding';
+        await profilesService[actionAPI](binding, {headers});
+        errIndex++;
+        const users = await getContributors(namespace, profilesService);
+        res.json(users);
+    } catch (err) {
+        const errMessage = [
+            `Unable to add new contributor for ${namespace}: ${err.stack || err}`,
+            `Unable to fetch contributors for ${namespace}: ${err.stack || err}`,
+        ][errIndex];
+        surfaceProfileControllerErrors({
+            res,
+            msg: errMessage,
+            err,
+        });
+    }
+};
+
+export const contributorAPI = (profilesService: DefaultApi) => Router()
+    .post('/create', async (req: Request, res: Response) => {
+        console.log('Recieved a call');
+        if (!req.user.hasAuth) {
+            return apiError({
+                res,
+                code: 405,
+                error: ERRORS.operation_not_supported,
+            });
+        }
+        
+        console.log('No Error');
```

Comment:
These seem like they were added for testing. I'd probably drop them in the final version.

Reply:
The only debug statements I forgot to add todo's for, thanks!

Reply:
Removed!